StyleCop.Analyzers/StyleCopTester/Program.cs

```diff
@@ -0,0 +1,129 @@
+
+            foreach (var analyzer in analyzers)
+            {
+                if (useAll || analyzer.SupportedDiagnostics.Any(y => ids.Contains(y.Id)))
+                {
+                    yield return analyzer;
+                }
+            }
+        }
+
+        private static ImmutableArray<DiagnosticAnalyzer> GetAllAnalyzers()
+        {
+            Assembly assembly = typeof(StyleCop.Analyzers.NoCodeFixAttribute).Assembly;
+
+            var diagnosticAnalyzerType = typeof(DiagnosticAnalyzer);
+
+            List<DiagnosticAnalyzer> analyzers = new List<DiagnosticAnalyzer>();
+
+            foreach (var type in assembly.GetTypes())
+            {
+                if (type.IsSubclassOf(diagnosticAnalyzerType) && !type.IsAbstract)
+                {
+                    analyzers.Add((DiagnosticAnalyzer)Activator.CreateInstance(type));
+                }
+            }
+
+            return analyzers.ToImmutableArray();
+        }
+
+        private static async Task<ImmutableList<Diagnostic>> GetAnalyzerDiagnosticsAsync(Solution solution, string solutionPath, ImmutableArray<DiagnosticAnalyzer> analyzers)
+        {
+            List<Task<ImmutableArray<Diagnostic>>> projectDiagnosticTasks = new List<Task<ImmutableArray<Diagnostic>>>();
+
+            // Make sure we analyze the projects in parallel
+            foreach (var project in solution.Projects)
+            {
+                projectDiagnosticTasks.Add(GetProjectAnalyzerDiagnostics(analyzers, project));
+            }
+
+            ImmutableList<Diagnostic>.Builder diagnosticBuilder = ImmutableList.CreateBuilder<Diagnostic>();
+            foreach (var task in projectDiagnosticTasks)
+            {
+                diagnosticBuilder.AddRange(await task.ConfigureAwait(false));
+            }
+
+            return diagnosticBuilder.ToImmutable();
+        }
+
+        /// <summary>
+        /// Returns a list of all analyzer diagnostics inside the specific project. This is an asynchronous operation.
+        /// </summary>
+        /// <param name="analyzers">The list of analyzers that should be used</param>
+        /// <param name="project">The project that should be analyzed</param>
+        /// <returns>A list of diagnostics inside the project</returns>
+        private static async Task<ImmutableArray<Diagnostic>> GetProjectAnalyzerDiagnostics(ImmutableArray<DiagnosticAnalyzer> analyzers, Project project)
+        {
+            Compilation compilation = await project.GetCompilationAsync().ConfigureAwait(false);
+            CompilationWithAnalyzers compilationWithAnalyzers = compilation.WithAnalyzers(analyzers);
+
+            return await compilationWithAnalyzers.GetAnalyzerDiagnosticsAsync().ConfigureAwait(false);
+        }
+
+        private static void PrintHelp()
+        {
+            Console.WriteLine("Usage: StyleCopTester [options] <Solution>");
```

Comment:
This should be `Console.WriteLine("Usage: StyleCopTester <Solution> [options]")`